### ISSUE TITLE
FLAGS,what's mean in the code?

### ISSUE THREAD

**Issue by cjr0106** [tensorflow/tensorflow#20676]
I could't understand the mean ''FLAGS" in the code,just like in your code:

`with tf.name_scope('train'):
  train_step = tf.train.AdamOptimizer(FLAGS.learning_rate).minimize(
      cross_entropy)

with tf.name_scope('accuracy'):
  with tf.name_scope('correct_prediction'):
    correct_prediction = tf.equal(tf.argmax(y, 1), tf.argmax(y_, 1))
  with tf.name_scope('accuracy'):
    accuracy = tf.reduce_mean(tf.cast(correct_prediction, tf.float32))
tf.summary.scalar('accuracy', accuracy)

# Merge all the summaries and write them out to /tmp/mnist_logs (by default)
merged = tf.summary.merge_all()
train_writer = tf.summary.FileWriter(FLAGS.summaries_dir + '/train',
                                      sess.graph)
test_writer = tf.summary.FileWriter(FLAGS.summaries_dir + '/test')`

FLAGS is None by default, so what's the  mean of FLAGS.learning_rate,FLAGS.summaries_dir?
or could you give me a example about the dir?

**Comment by lexee** [comment]
normally you will see code like ` tf.app.flags.DEFINE_string(...)` which defines some paras before training the model. Consider FLAGS as wrapperd  argparse in tensorflow.
Refer to [1] & [2] for more information.
[1] https://stackoverflow.com/questions/33932901/whats-the-purpose-of-tf-app-flags-in-tensorflow
[2] https://github.com/abseil/abseil-py

**Comment by cjr0106** [comment]
Thanks so much ,I got it~

2018-07-11 11:25:51lexee <notifications@github.com>写道：

normally you will see code like tf.app.flags.DEFINE_string(...) which defines some paras before training the model. Consider FLAGS as wrapperd argparse in tensorflow.
Refer to [1] & [2] for more information.
[1] https://stackoverflow.com/questions/33932901/whats-the-purpose-of-tf-app-flags-in-tensorflow
[2] https://github.com/abseil/abseil-py
